BaseTools/Source/C/Common/EfiUtilityMsgs.c

```diff
@@ -421,7 +421,8 @@ PrintMessage (
       strncpy (Line, Cptr, MAX_LINE_LEN - 1);
       Line[MAX_LINE_LEN - 1] = 0;
       if (LineNumber != 0) {
-        sprintf (Line2, "(%u)", (unsigned) LineNumber);
+        snprintf (Line2, MAX_LINE_LEN, "(%u)", (unsigned) LineNumber);
+        Line2[MAX_LINE_LEN-1] = '\0';

```

Comment:
The additional manual string nul termination appears to me to be both unnecessary and not reassuring in a security focused update.